controllers/volumereplicationgroup_controller.go

```diff
@@ -146,6 +158,53 @@ func (r *VolumeReplicationGroupReconciler) configMapFun(configmap client.Object)
 	return req
 }
 
+func (r *VolumeReplicationGroupReconciler) waitForCrds() error {
+	log := r.Log.WithName("checkingDependencies")
+
+	err := r.waitForResource(log, volumeReplicationClass)
+	if err != nil {
+		log.Error(err, "failed to wait for VolumeReplicationClass CRD")
+
+		return err
+	}
+
+	err = r.waitForResource(log, volumeReplication)
+	if err != nil {
+		log.Error(err, "failed to wait for VolumeReplication CRD")
+
+		return err
+	}
```

Comment:
This is little bit repetitive, we could make this more generic and easier to extend:

```Go
requiredCrds := []string{"VolumeReplicationClass", "VolumeReplication"}

for crd := range requiredCrds {
    err = r.waitForResource(log, crd)
    if err != nil {
        log.Error(err, "failed to wait for resource", "CRD", crd)
		return err
    }
}
```